### PR TITLE
SHOT-3556: Playlist and Version Experience for VRED Review

### PR DESCRIPTION
**Pull request by staceyoue**
* Simplified logic in `init_app` method
* Modified `_launch_via_hook` method to load in a Version or Playlist context in SG Panel when launching VRED for review
* Renamed config value from `viewer_extensions` to `accepted_published_file_types` and now specifies the PublishedFileType name instead of the file extension to allow easier filtering at the API level

@rob-aitchison I originally wanted to move the config option `accepted_published_file_types` to tk-vred config instead of this app, since the engine needs this value as well; however, I couldn't get this app to retrieve the config from tk-vred -- it was just empty, but was OK when accessing from tk-vred itself. Can you not access engine config settings from an app?

### CODE REVIEW

**Comment by rob-aitchison** [comment]
Hey @staceyoue, for clarity, are you trying to access the engine config setting before the engine is running?

**Comment by staceyoue** [comment]
@rob-aitchison I believe that is what is happening, we would need to access the config setting in the app.py method `_launch_via_hook`. Ideally tk-shotgun-launchvredreview and tk-vred could share the config settings `accepted_published_file_types`

**Comment by rob-aitchison** [comment]
@staceyoue , yes, agreed, let's try and figure out a way to use only the engine location for settings if possible.

File: app.py [hunk 1]
```diff
@@ -26,154 +26,222 @@ class ReviewWithVRED(Application):
...
-        )
-        path_on_disk = d.get("path").get("local_path")
-
-        # If this PublishedFile came from a zero config publish, it will
-        # have a file URL rather than a local path.
+        if published_file.get("error", None) is not None:
+            # There was an error getting the published file from the entity
+            raise TankError(published_file["error"])
+
+        # Extract the path on local disk for the published file that will be openedin VRED
```
Comment (rob-aitchison) [inline app.py:99]: typo - missing space openedin

**Review by rob-aitchison — CHANGES_REQUESTED** [review]
One typo spotted, otherwise all good, thanks!

**Review by rob-aitchison — DISMISSED** [review]
Thanks! 👍🏻

**Review by barbara-darkshot — DISMISSED** [review]
LGTM!

**Comment by barbara-darkshot** [comment]
hey @staceyoue and @rob-aitchison , I think this maybe help you to access tk-vred settings from this app even if the engine has not yet been started: https://github.com/shotgunsoftware/tk-alias/blob/master/engine.py#L488-L501
In my case, I'm accessing the settings from another project but you could use the same logic to get tk-vred settings and avoid introducing this duplicate setting inside of this app

**Comment by staceyoue** [comment]
> hey @staceyoue and @rob-aitchison , I think this maybe help you to access tk-vred settings from this app even if the engine has not yet been started: https://github.com/shotgunsoftware/tk-alias/blob/master/engine.py#L488-L501
> In my case, I'm accessing the settings from another project but you could use the same logic to get tk-vred settings and avoid introducing this duplicate setting inside of this app

Thank Babara! With this suggestion I was able to define the settings only in the tk-vred engine, and this app will be able to then access those settings, to avoid duplicating it.

**Review by rob-aitchison — APPROVED** [review]
Yes, looks great, thanks again @barbara-darkshot for the suggestion and @staceyoue for the quick turnaround! 👍🏻

**Pull request merged**